### DISCUSSION TITLE
Centering a println?

### DISCUSSION THREAD

**Discussion by krhatta** [Bodmer/TFT_eSPI#2831] [Q&A - General] [answered]
Hi! I'm fairly new to the Arduino world and coding in general, so I might come with some dumb questions, so I apologize in advance!

First of all, thank you Bodmer for the awesome library! I'm learning to use it right now and it's great. 

I am wondering if it's possible to center a tft.println int or float value? 
tft.drawCentreString works awesome, but I have some integers and floats that also need to be centered and right aligned. 

Thanks in advance!
Best, 
K

**Comment by Bodmer** [comment, marked as answer]
Use tft.setTextDatum(...) and the tft.drawString/drawFloat/drawNumber functions.

See examples.

**Reply by marclefevre** [reply]
Is it possible to this while using GFX fonts via the SetFreeFont() mechanism?  I find that setTextDatum doesn't seem to do anything if I'm using a Free font.

**Comment by krhatta** [comment]
Thanks a bunch!! I'll look at the examples and test it out.
